source/magpie/dataframe.d

```diff
@@ -1532,20 +1706,116 @@ public:
             return ret;
         }
     }
+
+    /++
+    Applies mathematical operations on DataFrame row/column
+    Params:
+        axis: 0 to compute row wise, 1 to compute column wise
+        Ops: Mathematical operations to apply
+    Returns:
+        DataFrame with computed operations  
+    +/
+    auto aggregate(int axis, Ops...)() @property
+    {
+        static if(axis)
```

Comment:
There is a lot of similar code in the two branches, try to push this static if deeper into the function.

Reply:
in particular do 
```
static if (axis)
{
    import std.meta: staticMap;
    alias Resolve(T) = suitableType!(resolverInternal!(T, Ops));
    alais DFTypes = staticMap!(Resolve, RowType);
}
else 
     alais DFTypes =aggregateType!(Ops)

DataFrame!(true, DFTypes) ret;
init!(axis)(ret);
...
```